src/runtime/hexagon/hexagon_vtcm_pool.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef TVM_RUNTIME_HEXAGON_HEXAGON_VTCM_POOL_H_
+#define TVM_RUNTIME_HEXAGON_HEXAGON_VTCM_POOL_H_
+
+#include <tvm/runtime/c_runtime_api.h>
+#include <tvm/runtime/device_api.h>
+#include <tvm/runtime/logging.h>
+#include <tvm/runtime/ndarray.h>
+#include <tvm/runtime/packed_func.h>
+
+#include <list>
+#include <utility>
+
+namespace tvm {
+namespace runtime {
+namespace hexagon {
+
+class HexagonVtcmPool {
```

Comment:
Can this be implemented in terms of `tvm::runtime::WorkspacePool` instead of requiring a separate pool manager?

* Pro: Avoid duplicate memory pool implementations
* Pro: `WorkspacePool` is designed around typical usage patterns for Relay models
* Con: `WorkspacePool` doesn't allow for different allocation scopes, would need to be added.
* Con: `WorkspacePool` doesn't perform any mutexing, assumes single threaded.
* Con: `WorkspacePool` only supports 1-d physical memory.  It looks like the current `HexagonVtcmPool` only supports 1-d physical memory, but keeping it separate gives more flexibility there for the future.

Given that those are some rather large cons, it makes sense to me either way, but I want to bring it up as an option.

Reply:
Thanks for raising this.  WorkspacePool will be used in the future, yes.  This change is to ensure that all allocations on the AllocDataSpace path will get buffers on the same page.  We accomplish this by allocating VTCM in one large chunk, and then managing buffers allocated and freed through Alloc/FreeDataSpace.